### ISSUE TITLE
Backspace in Input field causes webpage to navigate back to previous one

### ISSUE THREAD

**Issue by jrynkiew** [ocornut/imgui#4290]
Hi Ocurnut,

since a recent Firefox update the Backspace key navigates back to the previous webpage when pressed in the Input Text field.

This worked for a long time and I could successfully delete characters from the input text field. This is a recent change.

Delete still works (obviosly) as it doesn't have a native browser bind attached to it.

**Comment by PathogenDavid** [comment]
I can confirm this behavior in Firefox 89.0.2 (latest for Windows) when `browser.backspace_action` is set to `0` (which is no longer the default, but once upon a time it was. Not sure why you're seeing differently after an update.)

It appears SDL attempts to [prevent the default action of these keys](https://github.com/libsdl-org/SDL/blob/39302c921445e9f695a72469be17ba041f6f09db/src/video/emscripten/SDL_emscriptenevents.c#L571-L586). It's working as expected [in this random Emscripten+SDL2 demo I found](https://erik-larsen.github.io/emscripten-sdl2-ogles2/hello_triangle.html) as well as [the Dear ImGui interactive documentation prototype](https://pthom.github.io/imgui_manual_online/manual/imgui_manual.html) so it's not clear why it's not working in your app.

Have you done anything to manually change the keyboard event handlers for Emscripten or JavaScript directly?

**Comment by jrynkiew** [comment]
The only fix is to manually change the firefox option browser.backspace_action in about:config to "1" and save changes.
After doing this backspace works counter to space (space scrolls down, backspace scrolls up).

I don't think I can change this behaviour in the code of my app. Maybe some preventDefault somewhere?

**Comment by PathogenDavid** [comment]
> Are you saying that you also see this behaviour?

Yes, but only in your app. The issue does not happen with other SDL2 Emscripten apps.

> This is a specific bug in Firefox.

Not really. Once upon a time all browsers navigated back on backspace, there's ways to enable to old behavior on other browsers too. (For instance, I can reproduce the issue on Edgium by enabling the `#edge-backspace-key-navigate-page-back` flag.)

> browser.backspace_action in about:config to "1" and save changes.

You probably want to set it to `2` which disables backspace keybindings in general (which is the default on Windows Firefox and I assume others. Although I think distros are allowed to override the Mozilla defaults.)

> I don't think I can change this behaviour in the code of my app. Maybe some preventDefault somewhere?

SDL is already trying to do this, you should not have to add it. (See the other two apps I linked earlier, they do not have this issue.)

This strikes me more as an SDL or Emscripten question. You might have better luck asking in their respective communities. If you have not tried updating both SDKs it would be prudent to do so first.

**Comment by jrynkiew** [comment]
ok, I will look into it. Thanks. I will try to rebuild with new emscripten and SDL and see what happens. It's just surprising that it worked before the Firefox upgrade...From the documentation they said that the new Firefox was "dropping" the 'backspace to go back' functionality, and in my case upgrading to newer version actually caused it to happen, where at first is was woorking as intended (no backspace navigation)...

**Comment by jrynkiew** [comment]
> 
> This strikes me more as an SDL or Emscripten question. You might have better luck asking in their respective communities. If you have not tried updating both SDKs it would be prudent to do so first.

Thank you PathogenDavid,

You were right. I had to update emsdk and with it the bundled SDL2 implementation. 
After compiling, all goes back to normal and the backspace does not perform any action on the browser anymore. It only deletes characters from input fields as it should be.

Thank you again, 
best of luck,
Jeremi Rynkiewicz

**Comment by PathogenDavid** [comment]
Glad I could help!
